src/blog/0.3/linear-types-borrowing.vmd

```diff
@@ -0,0 +1,673 @@
+---
+title: What Vale Taught Me About Linear Types, Borrowing, and Memory Safety
+subtitle: And some ancient egyptian hieroglyphic slates!
+realm: blog
+date: May Z, 2023
+path: blog/linear-types-borrowing
+layout: annotated
+author: Evan Ovadia
+sponsor: me
+namespace: c-blog m-annotated
+---
+
+
+[# E: Make sense in general?]
+
+[# E: Too heavy on the egyptian stuff? Also, I drop off of it pretty early, think I need to tie it in more near the end, or not needed?]
+
+[# E: Too much comparison to Rust?]
+
+
+I've had some pretty surreal experiences while making Vale. [# And a lot of unexpected experiences. There was one time where I dove down an ancient history rabbit hole, just so I could add some flavor to an article about memory safety.] The weirdest was when I realized the hidden fundamental truth beneath a lot of the programming languages we use today.
+
+
+# The ancient writings that lit up a new field
+
+
+In July of 1799, a soldier was digging some foundations to expand a fortress in the Egyptian town of Rashid, when he stumbled upon an ancient-looking slate. He took it to his commanding officer, who realised that it could be important, or at the very least, valuable.
+
+
+<div><center><img src="/images/rosetta.jpg" style="max-height: 400px;"/><span style="opacity: .8"></span></center></div>
+
+
+The slate had writings in three languages, including ancient Egyptian hieroglyphs. Nobody knew how to read hieroglyphs yet, except for some oval shapes which are known to contain the names of kings and queens.
+
+The other two languages say the same thing as each other, and also mention those same kings and queens.
+
+
+The scholar Jean-Francois Champollion used these clues to eventually decipher the entire stone, and enabled us to finally understand ancient Egyptian hieroglyphs.
+
+
+I wonder what Champollion felt in that moment, when he had that entire slate translated for the first time.
+
+
+I'm no scholar, and I certainly didn't discover anything new. But I think I felt like Champollion did, when Vale showed me the hidden truth about how we can use linear types to completely eliminate Vale's memory safety costs, and led to the creation of the *linear-aliasing model*.
```

Comment:
Very long sentence, hard to parse. Suggest breaking it up:

I'm no scholar and linear types are not anything new, but I still felt a glimmer of what Champollion felt when Vale unearthed clues connecting linear types to memory safety. What hidden truth was revealed? We can use linear types to completely eliminate Vale's memory safety costs, a new paradigm called the *linear-aliasing model*.

Reply:
I'm no scholar and these concepts have been bouncing around awhile individually, but I still felt a glimmer of what Champollion felt when he connected two existing ideas to unlock something completely new! Vale showed me that we can use linear types to completely eliminate Vale's memory safety costs in a new paradigm called the linear-aliasing model.